src/PhpParser/AstResolver.php

```diff
@@ -318,9 +316,8 @@ public function parseFileNameToDecoratedNodes(string $fileName): array
             return $this->parsedFileNodes[$fileName] = [];
         }
 
-        $file = new File($fileName, FileSystem::read($fileName));
```

Comment:
file already read in previous parse get stmts before, we just need the stmts that already collected in previous line, no need to create new File object and re-read the file, just use the file path and decorate 👍